com.unity.netcode.gameobjects/Runtime/Transports/UTP/UnityTransport.cs

```diff
@@ -839,11 +848,22 @@ private float ExtractPacketLoss(NetworkConnection networkConnection)
             {
                 var sharedContext = (ReliableUtility.SharedContext*)sharedBuffer.GetUnsafePtr();
 
-                var packetReceived = (float)sharedContext->stats.PacketsReceived;
-                var packetDropped = (float)sharedContext->stats.PacketsDropped;
-                var packetLoss = packetReceived > 0 ? packetDropped / packetReceived : 0;
+                var packetReceivedDelta = (float)(sharedContext->stats.PacketsReceived - m_PacketLossCache.PacketsReceived);
+                var packetDroppedDelta = (float)(sharedContext->stats.PacketsDropped - m_PacketLossCache.PacketsDropped);
+
+                // There can be multiple update happening in a single frame where no packets have transitioned
+                // In those situation we want to return the last packet loss value instead of 0 to avoid invalid swings
+                if (packetDroppedDelta == 0 && packetReceivedDelta == 0)
+                {
+                    return m_PacketLossCache.PacketLoss;
```

Comment:
Would it make sense to return 0 in this case? It would mean that you don't need to store the previous value 🤔

Reply:
Just a question, maybe what you have is already the best approach